hyperstone/plugins/hooks/context.py

```diff
@@ -0,0 +1,23 @@
+from typing import Optional, TYPE_CHECKING
+
+from hyperstone.emulator import HyperEmu
+
+if TYPE_CHECKING:
+    from hyperstone.plugins.hooks.base import ActiveHook
+
+
+class Context:
+    def __init__(self):
+        self.emu: Optional[HyperEmu] = None
+        self.hook: Optional['ActiveHook'] = None
+
+
+class DictContext(Context, dict):
```

Comment:
Do we really need dict? Can we use UserDict instead?

Reply:
"[UserDict.UserDict](http://docs.python.org/library/userdict.html?#UserDict.UserDict) has no substantial added value since Python 2.2, since, as @gs mention, you can now subclass dict directly -- it exists only for backwards compatibility with Python 2.1 and earlier, when builtin types could not be subclasses. Still, it was kept in Python 3 (now in its proper place in the collections module) since, as [the docs](http://docs.python.org/3.1/library/collections.html?#collections.UserDict) now mention," (https://stackoverflow.com/questions/1392396/advantages-of-userdict-class)[Source]

What are the advantages of `UserDict` over `dict` in python3?